django/contrib/admin/views/autocomplete.py

```diff
@@ -24,12 +25,19 @@ def get(self, request, *args, **kwargs):
             return JsonResponse({'error': '403 Forbidden'}, status=403)
 
         self.term = request.GET.get('term', '')
-        self.paginator_class = self.model_admin.paginator
+        app_label = request.GET.get('app_label')
+        model_name = request.GET.get('model_name')
+        field_name = request.GET.get('field_name')
```

Comment:
The `field_name` has to be validated else a malicious user could set it to a sensitive field (e.g. `'password'` for `User.password`) and retrieve data they are not supposed to be allowed to see.

You can use `self.model_admin.to_field_allowed(request, field_name)`

e.g.

https://github.com/django/django/blob/f1c011f9592ef5dc99a4a3d7c54ad30fe0d722c5/django/contrib/admin/options.py#L1822-L1824

A test should be added to make sure this is appropriately validated.

Reply:
I added the check and a test to verify it can not be misused.